src/tye_lab_to_nwb/ast_ecephys/convert_session.py

```diff
@@ -0,0 +1,197 @@
+                )
+            )
+        )
+
+    if events_file_path:
+        event_names_mapping = {
+            0: "reward_stimulus_presentation",
+            1: "phototagging",
+            2: "shock_stimulus_presentation",
+            3: "reward_delivery",
+            4: "shock_relay",
+            5: "port_entry",
+            6: "neutral_stimulus_presentation",
+        }
+        read_kwargs = dict(event_names_mapping=event_names_mapping)
+        events_source_data = dict(file_path=str(events_file_path), read_kwargs=read_kwargs)
+        source_data.update(dict(Events=events_source_data))
+
+        events_column_mappings = dict(onset="start_time", offset="stop_time")
+        events_conversion_options = dict(column_name_mapping=events_column_mappings)
+        conversion_options.update(
+            dict(
+                Events=events_conversion_options,
+            )
+        )
+
+    # Add pose estimation (optional)
+    if sleap_file_path:
+        sleap_source_data = dict(Behavior=dict(file_path=str(sleap_file_path)))
+        source_data.update(sleap_source_data)
+        if video_file_path:
+            sleap_source_data["Behavior"].update(video_file_path=str(video_file_path))
+
+    # Add video (optional)
+    if video_file_path:
+        source_data.update(dict(Video=dict(file_paths=[str(video_file_path)])))
+
+    converter = AStEcephysNWBConverter(source_data=source_data)
+
+    # Add datetime to conversion
+    metadata = converter.get_metadata()
+
+    # Update default metadata with the editable in the corresponding yaml file
+    editable_metadata_path = Path(__file__).parent / "metadata" / "general_metadata.yaml"
+    editable_metadata = load_dict_from_file(editable_metadata_path)
+    metadata = dict_deep_update(metadata, editable_metadata)
+
+    if subject_metadata:
+        metadata = dict_deep_update(metadata, dict(Subject=subject_metadata))
+
+    if "session_id" not in metadata["NWBFile"]:
+        ecephys_folder_name = ecephys_recording_folder_path.name
+        session_id = ecephys_folder_name.replace(" ", "").replace("_", "-")
+        metadata["NWBFile"].update(session_id=session_id)
+
+    nwbfile_path = Path(nwbfile_path)
+    nwbfile_name = nwbfile_path.name
+    if stub_test:
+        nwbfile_path = nwbfile_path.parent / "nwb_stub" / nwbfile_name
+    nwbfile_path.parent.mkdir(parents=True, exist_ok=True)
+
+    # Run conversion
+    converter.run_conversion(
+        nwbfile_path=str(nwbfile_path),
+        metadata=metadata,
+        conversion_options=conversion_options,
+        overwrite=True,
+    )
+
+
+if __name__ == "__main__":
+    # Parameters for conversion
+    # The path that points to the folder where the OpenEphys (.continuous) files are located.
+    ecephys_folder_path = Path(
+        "/Volumes/t7-ssd/OpenEphys/3014_2018-06-26_14-21-37_illidanDiscD4",
+    )
```

Comment:
Can you design this mimicking their preferred approach from the last dataset where this takes a CSV as input? and use the example paths here as a default CSV

Reply:
Sure!

Reply:
Sure sure, sounds good.